### ISSUE TITLE
Internals update request

### ISSUE THREAD

**Issue by resistancelion** [Lymia/MPPatch#65]
Update internals for classes:
Modding
- (+) RescanModsFolder() or RefreshAvailableModsList()
as it says, refresh available mods - scan folder for subfolders/*.civ5mod files
- (+) RefreshSubscribedModsList()
scan for new available to download mods from previous scan or DB update
- (+) ResetDatabase()
clean mod cache database without erasing user settings (created by Modding.SetSystemProperty)
- (+) string GetCachePath()
return user-related game cache path

ModDetails (properties)
- (+) true|false SupportsMac
- (+) true|false SupportsHotSeat
- (+) string Categories
return categories to which it mod belongs, separated by commas

It will fix some bugs and add new filtering possibilities, but there's still one confusing bug with the mod database, which leads to another with gamesaves:
After mod's .modinfo file is edited, game recognizes the mod as new, which leads to:
1. The mod will be deactivated
2. The gamesave will no longer load, instead causing crash after attempt to load it with this mod
When testing i didn't edit mod id or name, so it must be connected with hash routines.

**Comment by Personwithhat** [comment]
This sounds good, would help some issues.
Maybe also automatically detect mods that require updates on start, and give some kind of a notification.
Have to bother people to check Mods>Scroll to mod > Check if 'Update' button is there

And then of course constantly hit issues such as:
- Subscribed to mod, but it doesn't download. Have to delete all other mods to get it to recognize it needs to be downloaded.
- Mod has had an update but update button doesn't appear

Random caching issue is my guess.

**Comment by resistancelion** [comment]
> This sounds good, would help some issues. Maybe also automatically detect mods that require updates on start, and give some kind of a notification. Have to bother people to check Mods>Scroll to mod > Check if 'Update' button is there
> 
> And then of course constantly hit issues such as:
> 
> * Subscribed to mod, but it doesn't download. Have to delete all other mods to get it to recognize it needs to be downloaded.
> * Mod has had an update but update button doesn't appear
> 
> Random caching issue is my guess.

Exactly, by the time i was writing it i was modifying  UI & functionality of the mods window, i've got it to work with filters, added ability to use & create modlist(modpack) files and  a few fixes.
But i do not reach my main goal - to make it work on Linux, Linux version of the game just reject any content that i try to add either as the mod or as the fake DLC, neither i could get to work regular modification or such of my own.

Also, @Lymia  can we discuss modding & your particular creation on any social network or by e-mail?
